hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManager.java

```diff
@@ -114,4 +115,12 @@ static RSGroupInfoManager create(MasterServices master) throws IOException {
    */
   void renameRSGroup(String oldName, String newName) throws IOException;
 
+  /**
+   * Update RSGroup configuration
+   * @param groupName the group name
+   * @param configuration new configuration of the group name to be set
+   * @throws IOException if a remote or network exception occurs
+   */
+  void updateRSGroupConfig(String groupName, Map<String, String> configuration) throws IOException;
+
```

Comment:
Similar question about the interface allowing deeper access to the map.

Reply:
I've changed the methods accessing the map in RSGroupInfo.
```
  public Map<String, String> getConfiguration() {
    return Collections.unmodifiableMap(configuration);
  }
  public void setConfiguration(String key, String value) {
    if (value == null) {
      removeConfiguration(key);
    } else {
      configuration.put(key, value);
    }
  }
  public void removeConfiguration(final String key) {
    configuration.remove(key);
  }
```
and rewrote the impl of RSGroupInfoManagerImpl.updateRSGroupConfig.